proto/v2/skaffold.proto

```diff
@@ -0,0 +1,819 @@
+message TriggerState {
+  oneof val {
+    bool enabled = 1; // enable or disable a trigger state
+  }
+}
+
+// Intent represents user intents for a given phase.
+message Intent {
+    bool build = 1; // in case skaffold dev is ran with autoBuild=false, a build intent enables building once
+    bool sync = 2; // in case skaffold dev is ran with autoSync=false, a sync intent enables file sync once
+    bool deploy = 3; // in case skaffold dev is ran with autoDeploy=false, a deploy intent enables deploys once
+}
+
+// Suggestion defines the action a user needs to recover from an error.
+message Suggestion {
+    SuggestionCode suggestionCode = 1; // code representing a suggestion
+    string action = 2; // action represents the suggestion action
+}
+
+// IntOrString is a type that can hold an int32 or a string.
+message IntOrString {
+    int32 type = 1; // type of stored value
+    int32 intVal = 2; // int value
+    string strVal = 3; // string value
+}
+
+// Describes all the methods for the Skaffold API
+service SkaffoldV2Service {
+
+    // Returns the state of the current Skaffold execution
+    rpc GetState (google.protobuf.Empty) returns (State) {
+        option (google.api.http) = {
+            get: "/v2/state"
+        };
+    }
+
+    // Returns all the events of the current Skaffold execution from the start
+    rpc Events (google.protobuf.Empty) returns (stream LogEntry) {
+        option (google.api.http) = {
+            get: "/v2/events"
+        };
+    }
+
+    // Allows for a single execution of some or all of the phases (build, sync, deploy) in case autoBuild, autoDeploy or autoSync are disabled.
+    rpc Execute (UserIntentRequest) returns (google.protobuf.Empty) {
+        option (google.api.http) = {
+            post: "/v2/execute"
+            body: "intent"
+        };
+    }
+
+    // Allows for enabling or disabling automatic build trigger
+    rpc AutoBuild (TriggerRequest) returns (google.protobuf.Empty) {
+        option (google.api.http) = {
+            put: "/v2/build/auto_execute"
+            body: "state"
+        };
+    }
+
+    // Allows for enabling or disabling automatic sync trigger
+    rpc AutoSync (TriggerRequest) returns (google.protobuf.Empty) {
+        option (google.api.http) = {
+            put: "/v2/sync/auto_execute"
+            body: "state"
+        };
+    }
+
+    // Allows for enabling or disabling automatic deploy trigger
+    rpc AutoDeploy (TriggerRequest) returns (google.protobuf.Empty) {
+        option (google.api.http) = {
+            put: "/v2/deploy/auto_execute"
+            body: "state"
+        };
+    }
+
+    // EXPERIMENTAL. It allows for custom events to be implemented in custom builders for example.
+    rpc Handle (Event) returns (google.protobuf.Empty) {
+        option (google.api.http) = {
+            post: "/v2/events/handle"
+            body: "*"
+        };
+    }
+
+}
+
+// Enum indicating builders used
+enum BuilderType {
```

Comment:
I was wondering if there is a way to keep enum for types and  error codes not version specific. That ways, as we add error codes, you dont have to add them in two places.

Reply:
I was also wondering about this. Maybe we should split enum types off into a separate file and import them in both?
https://developers.google.com/protocol-buffers/docs/proto3#importing_definitions

Reply:
I mentioned the UNRECOGNIZED issue we had with IntelliJ to @tstromberg and he mentioned that Google APIs generally uses a small set of high-level error codes and then a detail `reason` which is a string representations. See [AIP-193](https://google.aip.dev/193).  I wonder if we should just reuse those definitions.

Reply:
ok, I will make a follow up PR to address this 👍🏼